### PR TITLE
Add production asset compile

### LINKED ISSUE #24814: Production assets build should be tested in CI
### Pitch

The `assets:precompile` task broke multiple times in the past due to inconsistent `test` vs `production` settings.

For example:
- https://github.com/mastodon/mastodon/pull/24811
- https://github.com/mastodon/mastodon/pull/24374

We should have the CI run the assets precompile step, so those errors are caught before they are merged.

### Motivation

Currently we notice them because the docker image build fails, but this needs to be caught before.

### PR DESCRIPTION
**Pull request by nschonni**
Not sure this is excercising much else, but closes #24814

### CODE REVIEW

File: .github/workflows/test-ruby.yml [hunk 1]
```diff
@@ -16,6 +16,37 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  build-prod:
+    runs-on: ubuntu-latest
+
+    env:
+      RAILS_ENV: production
+
```
Comment (nschonni) [inline .github/workflows/test-ruby.yml:24]:
```suggestion
      BUNDLE_WITHOUT: 'development test'

```

**Comment by ClearlyClaire** [comment]
I'm not sure this should be under “Ruby Testing” since it's all about assets compilation, so more of a JS and toolchain thing.

The one thing I have against this is that it'll likely take a while to run, but if that does not significantly slows down the other testing jobs then that's not an issue.

**Review by ThisIsMissEm — COMMENTED** [review]
This looks reasonable, though, it may be possible to use a matrix for the environment variables which could reduce duplication here.

**Comment by nschonni** [comment]
Changed it to a matrix in a separate commit, so it can always be changed back

**Review by ClearlyClaire — APPROVED** [review]

**Review by renchap — APPROVED** [review]
👍

**Pull request merged**